src/helpers/pivot/pivot_time_adapter.ts

```diff
@@ -0,0 +1,174 @@
+    return (locale ?? DEFAULT_LOCALE).dateFormat;
+  },
+  formatValue(normalizedValue, locale) {
+    locale = locale ?? DEFAULT_LOCALE;
+    const value = toNumber(normalizedValue, DEFAULT_LOCALE);
+    return formatValue(value, { locale, format: this.getFormat(locale) });
+  },
+  toCellValue(normalizedValue) {
+    return toNumber(normalizedValue, DEFAULT_LOCALE);
+  },
+};
+
+/**
+ * Normalized value: "2/2023" for week 2 of 2023
+ */
+const weekAdapter: PivotTimeAdapter<string> = {
+  normalizeFunctionValue(value) {
+    const [week, year] = value.split("/");
+    return `${Number(week)}/${Number(year)}`;
+  },
+  getFormat() {
+    return undefined;
+  },
+  formatValue(normalizedValue) {
+    const [week, year] = normalizedValue.split("/");
+    return _t("W%(week)s %(year)s", { week, year });
+  },
+  toCellValue(normalizedValue) {
+    return this.formatValue(normalizedValue);
+  },
+};
+
+/**
+ * normalized month value is a string formatted as "MM/yyyy" (luxon format)
+ * e.g. "01/2020" for January 2020
+ */
+const monthAdapter: PivotTimeAdapter<string> = {
+  normalizeFunctionValue(value) {
+    const date = toNumber(value, DEFAULT_LOCALE);
+    return formatValue(date, { locale: DEFAULT_LOCALE, format: "mm/yyyy" });
+  },
+  getFormat() {
+    return "mmmm yyyy";
+  },
+  formatValue(normalizedValue, locale) {
+    locale = locale ?? DEFAULT_LOCALE;
+    const value = toNumber(normalizedValue, DEFAULT_LOCALE);
+    return formatValue(value, { locale, format: this.getFormat(locale) });
+  },
+  toCellValue(normalizedValue) {
+    return toNumber(normalizedValue, DEFAULT_LOCALE);
+  },
+};
+
+/**
+ * normalized quarter value is "quarter/year"
+ * e.g. "1/2020" for Q1 2020
+ */
+const quarterAdapter: PivotTimeAdapter<string> = {
+  normalizeFunctionValue(value) {
+    const [quarter, year] = value.split("/");
+    return `${quarter}/${year}`;
+  },
+  getFormat() {
+    return undefined;
+  },
+  formatValue(normalizedValue) {
+    const [quarter, year] = normalizedValue.split("/");
+    return _t("Q%(quarter)s %(year)s", { quarter, year });
+  },
+  toCellValue(normalizedValue) {
+    return this.formatValue(normalizedValue);
+  },
+};
+
+const yearAdapter: PivotTimeAdapter<number> = {
+  normalizeFunctionValue(value) {
+    return toNumber(value, DEFAULT_LOCALE);
+  },
+  getFormat() {
+    return "0";
+  },
+  formatValue(normalizedValue, locale) {
+    locale = locale ?? DEFAULT_LOCALE;
+    return formatValue(normalizedValue, { locale, format: "0" });
+  },
+  toCellValue(normalizedValue) {
+    return normalizedValue;
+  },
+};
+
+/**
+ * Decorate adapter functions to handle the empty value "false"
+ * @param {PivotTimeAdapter<string | number>} adapter
+ * @returns {PivotTimeAdapter<string | number | false>}
+ */
+function falseHandlerDecorator(
+  adapter: PivotTimeAdapter<string | number>
+): PivotTimeAdapter<string | number | false> {
+  return {
+    normalizeFunctionValue(value) {
+      if (value.toLowerCase() === "false") {
```

Comment:
The special `"false"` case is also a little funny in the context of o-spreadsheet :man_shrugging: